src/main/scala/nl/knaw/dans/easy/multideposit/actions/AddPropertiesToDeposit.scala

```diff
@@ -15,33 +15,78 @@
  */
 package nl.knaw.dans.easy.multideposit.actions
 
-import java.util.Properties
+import java.util
+import java.util.{Collections, Properties}
 
 import nl.knaw.dans.easy.multideposit.actions.AddPropertiesToDeposit._
-import nl.knaw.dans.easy.multideposit.{ Action, Settings, _ }
+import nl.knaw.dans.easy.multideposit.{Action, Settings, _}
 import resource._
 
 import scala.language.postfixOps
 import scala.util.control.NonFatal
-import scala.util.{ Failure, Success, Try }
+import scala.util.{Failure, Success, Try}
+import nl.knaw.dans.lib.error.TraversableTryExtensions
 
 case class AddPropertiesToDeposit(row: Int, entry: (DatasetID, Dataset))(implicit settings: Settings) extends Action {
 
   val (datasetID, dataset) = entry
 
   // TODO administratieve metadata, to be decided
 
-  /**
-   * @inheritdoc
-   * Checks whether there is only one unique DEPOSITOR_ID set in the `Dataset` (there can be multiple values but the must all be equal!).
-   * @return `Success` when all preconditions are met, `Failure` otherwise
-   */
-  override def checkPreconditions: Try[Unit] = validateDepositor(row, datasetID, dataset)
+  override def checkPreconditions: Try[Unit] = {
+    List(validateDepositor(row, datasetID, dataset), getDatamanagerMailadres(row))
+        .collectResults.map(_=>())
+  }
 
-  override def execute(): Try[Unit] = writeProperties(row, datasetID, dataset)
+  override def execute(): Try[Unit] = getDatamanagerMailadres(row).flatMap(email => writeProperties(row, datasetID, dataset, email))
 }
 object AddPropertiesToDeposit {
+  // The email needs to be acquired (from LDAP) only once during the program execution
+  private var datamanagerEmailaddress: Try[String] = null
+  // Only used for testing
+  private def resetDatamanagerEmailaddress() = {
+    datamanagerEmailaddress = null
+  }
+
+  /**
+   * Tries to retrieve the email address of the datamanager
+   * Also used for validation: checks if the datamanager is an active archivist with an email address
+   */
+  def getDatamanagerMailadres(row: Int)(implicit settings: Settings): Try[String] = {
+    // Note that the datamanager 'precondition' is checked when datamanagerEmailaddress is evaluated the first time
+    if(datamanagerEmailaddress == null) {
+      val id = settings.datamanager
+      datamanagerEmailaddress = settings.ldap.query(id)(a => a)
+        .flatMap(attrsSeq => {
+          if (attrsSeq.isEmpty) Failure(new ActionException(row, s"""DatamanagerID "$id" is unknown"""))
+          else if (attrsSeq.size > 1) Failure(new ActionException(row, s"""There appear to be multiple users with id "$id""""))
+          else Success(attrsSeq.head)
+        })
+        .flatMap(attrs => {
+          Option(attrs.get("dansState"))
+            .filter(_.get.toString == "ACTIVE")
+            .map(_ => Success(attrs))
+            .getOrElse(Failure(new ActionException(row, s"""The datamanager "$id" is not an active user""")))
```

Comment:
Now both the fact that `dansState` does not exist and the fact that the state is not equal to 'ACTIVE' will result in the same `Failure`. We discussed this, because you had an if-else here initially and you had convinced me that that was the better way. So I'm a bit surprised to see this here 😉 . Nevertheless I like this solution as well, so if you do not care about the two facts resulting in the same `Failure`, I'm OK with it!

Same for the two instances below